src/schematic/traits/autoscaler.rs

```diff
@@ -201,6 +202,10 @@ impl TraitImplementation for Autoscaler {
             match client.request::<hpa::HorizontalPodAutoscaler>(req) {
                 Ok(hpa) => hpa,
                 Err(e) => {
+                    if e.to_string().contains("NotFound") {
+                        warn!("Autoscaler not found {}. Recreating ...", e.to_string());
+                        self.add(ns, client).unwrap_or(());
```

Comment:
what if this function call returns an error？

Reply:
The reconciliation action doesn't happen but the error detected is propagated to the status field which is the current behavior.

Reply:
Can we add a unit test to cover this case?

Reply:
Sure, I need to mock some dependency to do it.